### ISSUE TITLE
Discussion: use class instead of exec for notification

### ISSUE THREAD

**Issue by faxm0dem** [camptocamp/puppet-systemd#2]
I _think_ it would make more sense to use this module like this:

```
file { '/lib/systemd/system/my.service':
   [...]
} ~> Class[Systemd::DaemonReload]
```

Reasons I can think of:
- make code be easier to modify without changing interface
- make code more portable: one dist may need executing something, the other noop

**Comment by tangseng5v** [comment]
Using class makes more sense. Agree .

**Comment by felixb** [comment]
:+1:

**Comment by faxm0dem** [comment]
Should we use a define or a class?
Which would make more sense (see also #5)

**Comment by faxm0dem** [comment]
After discussing this with @Volcane on IRC, it came out that it would probably make sense to go further with the abstraction, and to provide a `systemd::unit` defined type which would take care of everything.

**Comment by felixb** [comment]
@faxm0dem I've created a proposal PR for this already in #4.

**Comment by raphink** [comment]
https://github.com/camptocamp/puppet-systemd/issues/4 has been merged. I see from https://github.com/camptocamp/puppet-systemd/issues/5 that there was a suggestion still to use defines to avoid the problems described by @roidelapluie. Is that still a plan?

**Comment by reenberg** [comment]
Am I wrong in saying that this and #5 can be closed as this was implemented using a class in #18?

**Comment by faxm0dem** [comment]
I agree, thanks for noting this!
